common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java

```diff
@@ -1,473 +1,35 @@
-            Logger.info(TAG, "Sending intent to cancel authentication activity");
-
-            resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_CODE, parameters.get(ERROR));
-            resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_SUBCODE, parameters.get(ERROR_SUBCODE));
-
-            // Fallback logic on error_subcode when error_description is not provided.
-            // When error is "login_required", redirect url has error_description.
-            // When error is  "access_denied", redirect url has  error_subcode.
-            if (!StringUtil.isEmpty(parameters.get(ERROR_DESCRIPTION))) {
-                resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE, parameters.get(ERROR_DESCRIPTION));
-            } else {
-                resultIntent.putExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE, parameters.get(ERROR_SUBCODE));
-            }
-        } else {
-            Logger.info(TAG, "It is pointing to redirect. Final url can be processed to get the code or error.");
-            resultIntent.putExtra(AuthorizationStrategy.AUTHORIZATION_FINAL_URL, url);
-        }
-
-        return resultIntent;
-    }
-
-    /**
-     * Creates an intent to handle the completion of an authorization flow with browser. This restores
-     * the original AuthorizationActivity that was created at the start of the flow.
-     *
-     * @param context     the package context for the app.
-     * @param responseUri the response URI, which carries the parameters describing the response.
-     */
-    public static Intent createCustomTabResponseIntent(final Context context,
-                                                       final String responseUri) {
-        final Intent intent = new Intent(context, AuthorizationActivity.class);
-        intent.putExtra(AuthorizationStrategy.CUSTOM_TAB_REDIRECT, responseUri);
-        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
-        return intent;
-    }
-
-    private void extractState(final Bundle state) {
-        if (state == null) {
-            Logger.warn(TAG, "No stored state. Unable to handle response");
-            finish();
-            return;
-        }
+import com.microsoft.identity.common.R;
 
-        setDiagnosticContextForNewThread(state.getString(DiagnosticContext.CORRELATION_ID));
-        mAuthIntent = state.getParcelable(KEY_AUTH_INTENT);
-        mBrowserFlowStarted = state.getBoolean(KEY_BROWSER_FLOW_STARTED, false);
-        mPkeyAuthStatus = state.getBoolean(KEY_PKEYAUTH_STATUS, false);
-        mAuthorizationRequestUrl = state.getString(KEY_AUTH_REQUEST_URL);
-        mRedirectUri = state.getString(KEY_AUTH_REDIRECT_URI);
-        mRequestHeaders = getRequestHeaders(state);
-        mAuthorizationAgent = (AuthorizationAgent) state.getSerializable(KEY_AUTH_AUTHORIZATION_AGENT);
-    }
+public final class AuthorizationActivity extends FragmentActivity {
```

Comment:
`FragmentActivity` seems to be added in API 22, however the documentation says it's in AndroidX jetpack, did we test this on lower level devices (SDK 21 or below)?

Reply:
Verified with emulator SDK 16/21.